app/src/main/java/ch/epfl/qedit/view/Online/OnlineFragment.java

```diff
@@ -0,0 +1,201 @@
+                                    db.searchDatabase(load, load + 10, query).get())
+                                listAdapter.addItem(e);
+                            load = 10;
+                            searchView.clearFocus();
+                        } catch (Exception e) {
+                        }
+
+                        return false;
+                    }
+
+                    @Override
+                    public boolean onQueryTextChange(String newText) {
+                        load = 0;
+                        return false;
+                    }
+                });
+        super.onCreateOptionsMenu(menu, inflater);
+    }
+
+    // This method will be called when an item of the top bar is clicked on
+    @Override
+    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                requireActivity().onBackPressed();
+                break;
+        }
+
+        return true;
+    }
+
+    private CompletableFuture<StringPool> stringPool(String string) {
+        return db.getQuizLanguages(string)
+                .thenCompose(languages -> db.getQuizStringPool(string, getBestLanguage(languages)));
+    }
+
+    // Handles when a user clicked on the button to edit a quiz
+    private void editQuiz(int position) {
+        final String quizID = quizzes2.get(position).getKey();
+        progressBar.setVisibility(View.VISIBLE);
+
+        CompletableFuture<Quiz> quizStructure = db.getQuizStructure(quizID);
+
+        CompletableFuture.allOf(stringPool(quizID), quizStructure)
+                .whenComplete(
+                        (aVoid, throwable) -> {
+                            if (throwable != null)
+                                Toast.makeText(
+                                                requireContext(),
+                                                R.string.database_error,
+                                                Toast.LENGTH_SHORT)
+                                        .show();
+                            else
+                                launchActivity(
+                                        quizStructure
+                                                .join()
+                                                .instantiateLanguage(stringPool(quizID).join()));
+                        });
+    }
+
+    private String getBestLanguage(List<String> languages) {
+        String appLanguage = LocaleHelper.getLanguage(requireContext());
+
+        // If the quiz was translated in the application language, pick that version,
+        // otherwise we pick the first one. Note that we could have some more complex logic
+        // here, for example trying english first, and then falling back.
+        if (languages.contains(appLanguage)) return appLanguage;
+        else return languages.get(0);
+    }
+
+    private Intent save(Bundle b, Quiz q) {
```

Comment:
There is a small "bug": when we rotate the screen, the search string isn't saved, so we have to retype it. It would be nice to find a way to save it on a rotation change.